### PR TITLE
Fix #8999 - OutputControl:Files is marked unused

### LINKED ISSUE #8999: `OutputControl:Files` incorrectly produces "unused object" warning
Issue overview
--------------
When you include an `OutputControl:Files` object in your IDF, you get an incorrect warning from EnergyPlus:

```
   ************* There are 1 unused objects in input.
   ************* Use Output:Diagnostics,DisplayUnusedObjects; to see them.
```

If I use the Output:Diagnostics,DisplayUnusedObjects object:

```
   ** Warning ** The following lines are "Unused Objects".  These objects are in the input
   **   ~~~   **  file but are never obtained by the simulation and therefore are NOT used.
   **   ~~~   **  Only the first unused named object of an object class is shown.  Use Output:Diagnostics,DisplayAllWarnings; to see all.
   **   ~~~   **  See InputOutputReference document for more details.
   ************* Object=OutputControl:Files=OutputControl:Files 1
```

Example file to reproduce:
[in.idf.txt](https://github.com/NREL/EnergyPlus/files/7034510/in.idf.txt)

### Details
Some additional details for this issue (if relevant):
 - Windows 10
 - EnergyPlus v9.5

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added: https://github.com/NREL/EnergyPlusDevSupport/blob/master/DefectFiles/8000s/8999/in.idf
 - [x] Ticket added to Pivotal for defect (development team task)
 - [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------
 - Fixes #8999 
 - mark `OutputControl:Files` as used
 - Write a functional unit test for the OutputControlFiles object

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [x] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [x] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [x] If structural output changes, add to output rules file and add OutputChange label
 - [x] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: tst/EnergyPlus/unit/OutputFiles.unit.cc [hunk 1]
```diff
@@ -219,4 +219,244 @@ TEST_F(EnergyPlusFixture, OutputFiles_Expected_Formatting_Tests)
...
+        "   **   ~~~   **  file but are never obtained by the simulation and therefore are NOT used.",
+        "   **   ~~~   **  Only the first unused named object of an object class is shown.  Use Output:Diagnostics,DisplayAllWarnings; to see all.",
+        "   **   ~~~   **  See InputOutputReference document for more details.",
+        "   ************* Object=Building=Bldg",
+        "   **   ~~~   ** Object=GlobalGeometryRules",
+        "   **   ~~~   ** Object=Version",
+    });
+
+    compare_err_stream(expected_error);
+}
```
Comment (jmarrec) [inline tst/EnergyPlus/unit/OutputFiles.unit.cc:283]: Specific test for #8999

File: tst/EnergyPlus/unit/OutputFiles.unit.cc [hunk 2]
```diff
@@ -219,4 +219,244 @@ TEST_F(EnergyPlusFixture, OutputFiles_Expected_Formatting_Tests)
...
+  {wrl},              !- Output WRL
+  {screen},           !- Output Screen
+  {extshd},           !- Output ExtShd
+  {tarcog};           !- Output Tarcog
+  )";
+
+    auto boolToString = [](bool b) { return b ? "Yes" : "No"; };
+
+    for (int i = 0; i < 31; ++i) {
+        bool csv = (i == 0);
```
Comment (jmarrec) [inline tst/EnergyPlus/unit/OutputFiles.unit.cc:325]: Test every possible combination with just one as "True" to ensure everything is captured correctly in getInput

File: src/EnergyPlus/IOFiles.cc [hunk 3]
```diff
@@ -344,6 +344,8 @@ void IOFiles::OutputControl::getInput(EnergyPlusData &state)
         for (auto instance = instancesValue.begin(); instance != instancesValue.end(); ++instance) {
             auto const &fields = instance.value();
 
+            state.dataInputProcessing->inputProcessor->markObjectAsUsed("OutputControl:Files", instance.key());
```
Comment (jmarrec) [inline src/EnergyPlus/IOFiles.cc:347]: actual fix for #8999

**Comment by jmarrec** [comment]
Defect file on DevSupport https://github.com/NREL/EnergyPlusDevSupport/blob/master/DefectFiles/8000s/8999/in.idf, side by side comparison of eplusout.err before and after fix

image

File: src/EnergyPlus/IOFiles.cc (hunk 3 above)
Comment (Myoldmopar) [reply src/EnergyPlus/IOFiles.cc:347]: :+1:

File: tst/EnergyPlus/unit/OutputFiles.unit.cc (hunk 1 above)
Comment (Myoldmopar) [reply tst/EnergyPlus/unit/OutputFiles.unit.cc:283]: Nice.

File: tst/EnergyPlus/unit/OutputFiles.unit.cc [hunk 4]
```diff
@@ -219,4 +219,244 @@ TEST_F(EnergyPlusFixture, OutputFiles_Expected_Formatting_Tests)
...
+        state->files.outputControl.dbg = false;
+        state->files.outputControl.perflog = false;
+        state->files.outputControl.sln = false;
+        state->files.outputControl.sci = false;
+        state->files.outputControl.wrl = false;
+        state->files.outputControl.screen = false;
+        state->files.outputControl.extshd = false;
+        state->files.outputControl.tarcog = false;
+    }
+}
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/OutputFiles.unit.cc:460]: Wow.  Very nice.

**Review by Myoldmopar — APPROVED** [review]
Very nice, simple fix and a great set of unit tests!

**Comment by Myoldmopar** [comment]
CI is showing this is working great, with ERR diffs showing the error files have those messaged cleaned out.  No reason to wait long on this one.  Thanks @jmarrec !

**Pull request merged**